angular-ngrx-scss/src/app/repository/services/repository.service.ts

```diff
@@ -0,0 +1,79 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { map, Observable } from 'rxjs';
+import {
+  ReadmeApiResponse,
+  RepoApiResponse,
+  RepoContents,
+  RepoContentsApiResponse,
+  RepoState,
+} from 'src/app/state/repository';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RepositoryService {
+  constructor(private http: HttpClient) {}
+
+  getRepositoryInfo(owner: string, repoName: string): Observable<RepoState> {
+    const url = `${environment.githubUrl}/repos/${owner}/${repoName}`;
+
+    return this.http.get<RepoApiResponse>(url).pipe(
+      map((data) => ({
+        repoName: data.name,
+        description: data.description,
+        website: data.homepage,
+        visibility: data.visibility,
+        watchCount: data.watchers_count,
+        starCount: data.stargazers_count,
+        forkCount: data.forks_count,
+        issueCount: data.open_issues_count,
+        tags: data.topics,
+        ownerName: '',
+        prCount: 0,
+        readme: '',
+        tree: [],
+      })),
+    );
+  }
+
+  // TODO: set this method up to return the data as well as the count (issue #185)
+  // TODO: write test for this function when it's updated
+  getPullRequestList(owner: string, repoName: string): Observable<number> {
+    const url = `${environment.githubUrl}/repos/${owner}/${repoName}/pulls`;
+
+    return this.http.get<[]>(url).pipe(map((data) => data.length));
+  }
+
+  getRepositoryContents(
+    owner: string,
+    repoName: string,
+    path?: string,
+  ): Observable<RepoContents[]> {
+    const url = path
+      ? `${environment.githubUrl}/repos/${owner}/${repoName}/contents/${path}`
+      : `${environment.githubUrl}/repos/${owner}/${repoName}/contents`;
+
+    return this.http.get<RepoContentsApiResponse[]>(url).pipe(
+      map((data) => {
+        const files = data.map((value) => ({
+          name: value.name,
+          type: value.type,
+        }));
+
+        return files;
+      }),
+    );
+  }
+
+  // TODO: readme file is currently an encoded string - this method should be improved to either return the raw data, or include the fields needed to de-code the string when ready to display it
+  // TODO: write test for this function when it's updated
+  getReadmeContent(owner: string, repoName: string): Observable<string> {
+    const url = `${environment.githubUrl}/repos/${owner}/${repoName}/readme`;
+
+    return this.http
+      .get<ReadmeApiResponse>(url)
+      .pipe(map((file) => file.content));
+  }
```

Comment:
Do we know if any of the other starters are using a specific README rendering library? I am curious if a decision was already made at some point as to which one we should use.

Reply:
Not sure. Since this ticket is just setting up the API and not worrying about rendering the document I left it like this, but would be worth looking into and adjusting when it comes time to actually show the README! There has to be something, I'd think.